### PR TITLE
Update the unqualified-search-registries enhancement

### PR DESCRIPTION
**Pull request by umohnani8**
After a discussion on the risks that the use of short names
pose, we agreed on not adding support for short names to
builds and imagestream imports. Update the enhancement
accordingly.

Signed-off-by: Urvashi Mohnani <umohnani@redhat.com>

### CODE REVIEW

**Comment by umohnani8** [comment]
@bparees @dmage @mrunalp PTAL

File: enhancements/api-review/add-unqualified-search-registries.md [hunk 1]
```diff
@@ -132,15 +131,20 @@ Note: adding a drop-in file at `/etc/containers/registries.conf.d` completely ov
...
-Also, CRI-O will be doing the image search and will have access to the credentials for all the registries in the list.
+The `unqualified-search-registries` field already exists in the `/etc/containers/registries.conf` file on all nodes in a cluster. This enhancement just adds the ability for the user to be able to configure this list. An API change for adding the `UnqualifiedSearchRegistry` field to the images.config.openshift.io CR is needed and updates to the MCO code to handle this new option will be needed..
+CRI-O will be doing the image search and will be using the cluster wide pull secret to get the credentials for all the registries in the list.
+
+We will document a big warning about the security risks of using short names. This includes the following:
+
+- Users are subject to network and reistry-originating attacks when using external registries.
+- If the pod pull secret contains docker.io credentials, these credentials will be sent to all the registries in the search list.
+- Only the cluster-wide pull secret can be used for credentials.
```
Comment (bparees) [inline enhancements/api-review/add-unqualified-search-registries.md:145]: (plus the namespace'd docker.io creds, if present)

File: enhancements/api-review/add-unqualified-search-registries.md [hunk 2]
```diff
@@ -132,15 +131,20 @@ Note: adding a drop-in file at `/etc/containers/registries.conf.d` completely ov
...
-Also, CRI-O will be doing the image search and will have access to the credentials for all the registries in the list.
+The `unqualified-search-registries` field already exists in the `/etc/containers/registries.conf` file on all nodes in a cluster. This enhancement just adds the ability for the user to be able to configure this list. An API change for adding the `UnqualifiedSearchRegistry` field to the images.config.openshift.io CR is needed and updates to the MCO code to handle this new option will be needed..
+CRI-O will be doing the image search and will be using the cluster wide pull secret to get the credentials for all the registries in the list.
+
+We will document a big warning about the security risks of using short names. This includes the following:
+
+- Users are subject to network and reistry-originating attacks when using external registries.
+- If the pod pull secret contains docker.io credentials, these credentials will be sent to all the registries in the search list.
```
Comment (bparees) [inline enhancements/api-review/add-unqualified-search-registries.md:144]: how does crio choose between the cred the kubelet passed down, and the node creds that are available?

Comment (umohnani8) [reply enhancements/api-review/add-unqualified-search-registries.md:144]: When we get a pull secret from the kubelet we give it priority and send it over to the c/image library, I think there it prepends the registries from the search list to the short name image we are given and tries pulling the image from those registries with the kubelet pull secret. CRI-O also sends down the auth file to use to the c/image library, so if the pull fails with the kubelet secret it falls back to the auth file, which is the cluster wide pull secret from /var/lib/kubelet/config.json.

File: enhancements/api-review/add-unqualified-search-registries.md (hunk 1 above)
Comment (umohnani8) [reply enhancements/api-review/add-unqualified-search-registries.md]: Fixed.

File: enhancements/api-review/add-unqualified-search-registries.md [hunk 3]
```diff
@@ -46,6 +46,8 @@ Users can currently configure `AllowedRegistries`, `BlockedRegistries`, and `Ins
 
 ### Non-Goals
 
+- Builds and Imagestream Imports will not support the use of short names.
```
Comment (mrunalp) [inline enhancements/api-review/add-unqualified-search-registries.md:49]: Could you add justification for why we decided on this?

Comment (umohnani8) [reply enhancements/api-review/add-unqualified-search-registries.md]: Added

**Pull request merged**